src/adapters/mqtt/index.ts

```diff
@@ -20,105 +19,157 @@ class MqttAdapter extends Adapter {
   }
 
   async connect(): Promise<this> {
-    return this._connect()
+    return await this._connect()
```

Comment:
You don't have to `await` it:

```suggestion
    return this._connect()
```